pkg/external-provider/errors.go

```diff
@@ -0,0 +1,21 @@
+package provider
+
+import "errors"
+
+var (
+	// ErrNewOperatorNotSupportedByPrometheus creates an error that represents the fact that we were requested to service a query that
+	// Prometheus would be unable to support.
+	ErrNewOperatorNotSupportedByPrometheus = errors.New("operator not supported by prometheus")
+
+	// ErrNewOperatorRequiresValues creates an error that represents the fact that we were requested to service a query
+	// that was malformed in its operator/value combination.
+	ErrNewOperatorRequiresValues = errors.New("operator requires values")
+
+	// ErrNewOperatorDoesNotSupportValues creates an error that represents the fact that we were requested to service a query
+	// that was malformed in its operator/value combination.
+	ErrNewOperatorDoesNotSupportValues = errors.New("operator does not support values")
```

Comment:
from the comment there seems to be no difference to `ErrNewOperatorRequiresValues`. Can we be more precise in the godoc?

naming nit: `ErrMalformedQueryUnsupportedValues`